cms/core/utils.py

```diff
@@ -1,10 +1,18 @@
+import io
 from typing import TYPE_CHECKING, Any, Optional, TypedDict
 
+import matplotlib as mpl
```

Comment:
Issue: matplotlib is not thread safe: https://matplotlib.org/stable/users/faq.html#work-with-threads.

Is there a different library we could use instead for this? Alternatively, this should be wrapped in a thread lock.

Reply:
Thanks for flagging this, I wrapped it in a lock and refactored it slightly based on [this](https://matplotlib.org/stable/gallery/user_interfaces/web_application_server_sgskip.html).

[1eda5ab](https://github.com/ONSdigital/dis-wagtail/pull/192/commits/1eda5abd109e35a9c8609d3703d04f707c4e2f33)